### PR TITLE
feat(issue-details): Add editing for comments in streamlined experience

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr adds back editing for activity comments in the streamlined experience. this existed in the past but didn't make it in the initial version of the activity side. 

Screenshot 2024-12-05 at 1 50 03 PM

### CODE REVIEW

File: static/app/views/issueDetails/streamline/sidebar/activitySection.tsx [hunk 1]
```diff
@@ -82,7 +89,19 @@ function TimelineItem({
...
+          itemKey={item.id}
+          storageKey={`groupinput:${item.id}`}
+          source="issue-details"
+          text={item.data.text}
+          noteId={item.id}
+          onUpdate={n => {
+            handleUpdate(item, n);
+            setEditing(false);
+          }}
+        />
```
Comment (leeandher) [inline static/app/views/issueDetails/streamline/sidebar/activitySection.tsx:103]: Could we have the edit button have a Save/Cancel button? Currently there's no way back from editing without a page refresh.

**Review by leeandher — COMMENTED** [review]
Looks slick! I think changing the text to `Save` or `Update` should be straightforward but hopefully its not too tricky to add a cancel button.

Bonus points if you can cancel by clicking away or pressing Esc

File: static/app/views/issueDetails/streamline/sidebar/activitySection.tsx (hunk 1 above)
Comment (roggenkemper) [reply static/app/views/issueDetails/streamline/sidebar/activitySection.tsx]: yup - will add a cancel button!

**Review by scttcper — APPROVED** [review]
great!

**Pull request merged**